### PR TITLE
Register HomeWorkspace events upon JSON deserialization

### PR DESCRIPTION
**Pull request by alfarok**
### Purpose

[QNTM-2124](https://jira.autodesk.com/browse/QNTM-2124)

The purpose of this PR is to register the `HomeWorkspace` with the `OnEvaluationCompleted` and `OnRefreshCompleted` events.  This has always been the case previously with XML and was accidentally omitted when JSON deserilaization was added.  This also caused some buggy behavior, such as Dynamo not properly communicating/updating with Revit.

### Testing

Added a unit test for verifying the  `OnEvaluationCompleted` and `OnRefreshCompleted` events are registered with the homeworkspace upon xml and json deserialization.

### Declarations

Check these if you believe they are true

- [X] The code base is in a better state after this PR
- [X] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [X] The level of testing this PR includes is appropriate
- [X] User facing strings, if any, are extracted into `*.resx` files
- [X] All tests pass using the self-service CI.
- [X] Snapshot of UI changes, if any.
- [X] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@mjkkirschner  @QilongTang 

### FYIs

@smangarole

### CODE REVIEW

File: test/DynamoCoreWpfTests/WorkspaceOpeningTests.cs [hunk 1]
```diff
@@ -20,6 +21,63 @@ public void OpeningWorkspaceSetsPosition()
         }
 
         [Test]
+        public void VerifyRegisteredHomeWorkspace()
+        {
+            // This test verifies the HomeWorkspace was registered with the EvaluationCompleted
+            // and RefreshCompleted events upon deserialization in both xml and json
+
```
Comment (alfarok) [inline test/DynamoCoreWpfTests/WorkspaceOpeningTests.cs]: this test fails if the HomeWorkspace does not get registered (as shown above)

File: test/DynamoCoreWpfTests/WorkspaceOpeningTests.cs [hunk 2]
```diff
@@ -20,6 +21,63 @@ public void OpeningWorkspaceSetsPosition()
...
+            
+            // Verify events handlers have been registered
+            bool openXmlFired = false;
+            ViewModel.Model.EvaluationCompleted += (sender, e) => openXmlFired = true;
+            RunCurrentModel();
+            Assert.IsTrue(openXmlFired);
+
+            //Unsubscribe
+            ViewModel.Model.EvaluationCompleted -= ViewModel.Model.OnEvaluationCompleted;
+
```
Comment (alfarok) [inline test/DynamoCoreWpfTests/WorkspaceOpeningTests.cs]: @mjkkirschner not sure if I am doing this correctly?

File: test/DynamoCoreWpfTests/WorkspaceOpeningTests.cs [hunk 3]
```diff
@@ -20,6 +21,63 @@ public void OpeningWorkspaceSetsPosition()
...
+            // Close workspace
+            Assert.IsTrue(ViewModel.CloseHomeWorkspaceCommand.CanExecute(null));
+            ViewModel.CloseHomeWorkspaceCommand.Execute(null);
+
+            // Delete temp file
+            File.Delete(tempPath);
+
+            //Unsubscribe
+            ViewModel.Model.EvaluationCompleted -= ViewModel.Model.OnEvaluationCompleted;
```
Comment (mjkkirschner) [inline test/DynamoCoreWpfTests/WorkspaceOpeningTests.cs]: this is the wrong this to unsubscribe, you need to unsubscribe your lambda functions above, to unsubscribe these you'll most likely need to name them or assign them to a variable.

**Comment by mjkkirschner** [comment]
last thing is style preference, I prefer braces around functions, can you check the style guide in the wiki to see if thats codified or not? If it's not - LGTM.

**Comment by alfarok** [comment]
added brackets to the lambda expressions

**Comment by QilongTang** [comment]
LGTM

**Pull request merged**